dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/jaxrspropagation/common/PropagationHeaderEndpoint.java

```diff
@@ -0,0 +1,79 @@
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.microprofile.telemetry.internal_fat.apps.jaxrspropagation.common;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.baggage.BaggageEntry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.trace.Span;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.ws.rs.ApplicationPath;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.Application;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.HttpHeaders;
+
+@ApplicationPath("/")
+@Path("/propagationHeaderEndpoint")
+public class PropagationHeaderEndpoint extends Application {
+
+    public static final String BAGGAGE_KEY = "test.baggage.key";
+    public static final AttributeKey<String> BAGGAGE_METADATA_ATTR = AttributeKey.stringKey("test.baggage.metadata");
+    public static final AttributeKey<String> BAGGAGE_VALUE_ATTR = AttributeKey.stringKey("test.baggage");
+    public static final AttributeKey<List<String>> PROPAGATION_HEADERS_ATTR = AttributeKey.stringArrayKey("test.propagation.headers");
+
+    private static final List<String> PROPAGATION_HEADER_NAMES = Arrays.asList("baggage", "traceparent",
+                                                                               "b3",
+                                                                               "X-B3-TraceId", "X-B3-SpanId", "X-B3-ParentSpanId", "X-B3-Sampled",
+                                                                               "uber-trace-id");
+
+    @GET
+    public String get(@Context HttpHeaders headers) {
+        Span span = Span.current();
+
+        // Extract the propagation headers and store in the span
+        List<String> propagationHeaders = headers.getRequestHeaders().keySet().stream()
+                        .filter(PropagationHeaderEndpoint::isPropagationHeader)
+                        .collect(Collectors.toList());
+        span.setAttribute(PropagationHeaderEndpoint.PROPAGATION_HEADERS_ATTR, propagationHeaders);
+
+        // Extract the test baggage value (if present) and store in the span
+        BaggageEntry baggageEntry = Baggage.current().asMap().get(BAGGAGE_KEY);
+        if (baggageEntry != null) {
+            span.setAttribute(BAGGAGE_VALUE_ATTR, baggageEntry.getValue());
+            span.setAttribute(BAGGAGE_METADATA_ATTR, baggageEntry.getMetadata().getValue());
+        }
+
+        return "OK";
+    }
```

Comment:
Shouldn't there be some assertions in here? I don't think the returned value is ever checked either?

Reply:
I've tried to avoid doing assertions in here since it's a bunch of work to get the `AssertionError` propagated back to the test servlet and hence the junit results.

Aside: I did consider trying to create something to do this, I don't think it should be too hard, we would extract the serialization logic from the FATServlet and make it into something that can be used by JAX-RS.

Instead here I've taken the easier option of setting the values we want to check into the span. Since the test method has to collect the spans to check they're parented correctly, it's easy to assert the attributes that `PropagationHeaderEndpoint` sets.

Possibly I should have done a better job of documenting what `PropagationHeaderEndpoint` does and how tests should use it.

Reply:
Added class javadoc to `PropagationHeaderEndpoint` to document what it does and how it's intended to be used.

Reply:
ok